torch/csrc/inductor/aot_inductor_tensor.cpp

```diff
@@ -0,0 +1,185 @@
+      C10DeviceType_to_AotInductorDeviceType.count(device.type()) != 0,
+      "DeviceType not supported by AotInductor");
+  AotInductorDevice result{
+      C10DeviceType_to_AotInductorDeviceType.at(device.type()), device.index()};
+  return result;
+}
+
+c10::Device convert_to_c10_device(AotInductorDevice device) {
+  TORCH_CHECK(
+      AotInductorDeviceType_to_C10DeviceType.count(device.device_type) != 0,
+      "DeviceType not supported by AotInductor");
+  c10::Device result{
+      AotInductorDeviceType_to_C10DeviceType.at(device.device_type),
+      device.device_id};
+  return result;
+}
+
+std::vector<std::vector<int64_t>> sizes_cache;
+std::vector<std::vector<int64_t>> strides_cache;
+
+const int64_t* register_sizes(c10::IntArrayRef sizes) {
+  sizes_cache.emplace_back(sizes.vec());
+  return sizes_cache[sizes_cache.size() - 1].data();
+}
+
+const int64_t* register_strides(c10::IntArrayRef strides) {
+  strides_cache.emplace_back(strides.vec());
+  return strides_cache[strides_cache.size() - 1].data();
+}
+
+} // namespace
+
+AotInductorTensor convert_to_aot_inductor_tensor(void* aten_tensor) {
+  at::Tensor* t = static_cast<at::Tensor*>(aten_tensor);
+
+  AotInductorTensor result = {
+      t->data_ptr(),
+      convert_to_aot_inductor_device(t->device()),
+      convert_to_aot_inductor_scalar_type(t->scalar_type()),
+      t->dim(),
+      register_sizes(t->sizes()),
+      register_strides(t->strides())};
+  return result;
+}
+
+void convert_to_aten_tensor(
+    AotInductorTensor inductor_tensor,
+    void* aten_tensor) {
+  c10::TensorOptions options =
+      c10::TensorOptions()
+          .device(convert_to_c10_device(inductor_tensor.device))
+          .dtype(convert_to_c10_scalar_type(inductor_tensor.type));
+
+  at::Tensor* t = static_cast<at::Tensor*>(aten_tensor);
+  (*t) = at::from_blob(
+      inductor_tensor.data_ptr,
+      c10::IntArrayRef(inductor_tensor.sizes, inductor_tensor.ndim),
+      c10::IntArrayRef(inductor_tensor.strides, inductor_tensor.ndim),
+      options);
+}
+
+AotInductorTensor aot_inductor_empty_strided(
+    int64_t dim,
+    int64_t* sizes_ptr,
+    int64_t* strides_ptr,
+    AotInductorDevice device,
+    AotInductorScalarType type) {
+  c10::ArrayRef sizes(sizes_ptr, dim);
+  c10::ArrayRef strides(strides_ptr, dim);
+  at::Tensor result = at::empty_strided(
+      sizes,
+      strides,
+      c10::TensorOptions(convert_to_c10_device(device))
+          .dtype(convert_to_c10_scalar_type(type)));
+  return convert_to_aot_inductor_tensor(&result);
+}
+
+AotInductorTensor aot_inductor_as_strided(
+    AotInductorTensor self,
+    int64_t* sizes_ptr,
+    int64_t* strides_ptr,
+    int64_t offset) {
+  c10::ArrayRef sizes(sizes_ptr, self.ndim);
+  c10::ArrayRef strides(strides_ptr, self.ndim);
+  at::Tensor self_tensor;
+  convert_to_aten_tensor(self, &self_tensor);
+  at::Tensor result = at::as_strided(self_tensor, sizes, strides, offset);
+  return convert_to_aot_inductor_tensor(&result);
```

Comment:
We can implement this directly on AotInductorTensor and get much better perf than calling at::as_strided.